### DISCUSSION TITLE
react-navigation-visualizer cli — automatically generate a "sitemap" of Screens

### DISCUSSION THREAD

**Discussion by flexbox** [react-navigation/react-navigation#9810] [General]
I joined a project and the architecture of `StackNavigator`s is a bit tricky to understand.

Is it possible to generate the structure / a mindmap of all Screens components inside a `NavigationContainer`?

**I am looking for a cli tool to generate a sitemap of screens**

Example:
```
yarn run react-native-bundle-visualizer      <---- this one is for the bundle size
```

Or do I need to reproduce the user flow by hand with:
- https://flowchart.fun
- https://excalidraw.com

**Comment by gianpaj** [comment]
Something like this would be great

At the moment I'm using React Native Debugger to see the entire React tree

![Screen Shot 2021-09-05 at 13 57 48](https://user-images.githubusercontent.com/899175/132127676-0eaba45d-7d68-4faf-995f-3a343c77a1b1.png)

if use a regex to find "Stack" or "Navigation" components it doesn't show in a tree shape

![image](https://user-images.githubusercontent.com/899175/132127781-17aab8c8-e955-4ba2-be56-47b679be4bea.png)

**Reply by flexbox** [reply]
@gianpaj This is nice thanks for sharing!

**Comment by thomasrebam** [comment]
Hey ! This discussion is quite old, but if this can be useful to anyone, I extended the expo dev-plugins to add a navigation visualizer : https://www.npmjs.com/package/@bam.tech/react-navigation-visualizer-dev-plugin.

The visualization looks like this:
![image](https://github.com/user-attachments/assets/57badbc4-4207-48cd-837d-4a4dff344383)
The stacks and screen are represented as an actual stack, not exactly a sitemap, but it can already help visualize or communicate around navigation 😄

**Reply by flexbox** [reply]
Thanks @thomasrebam your plugin looks really useful.
I will test it right now.
